lib/web_ui/lib/src/engine/canvas_pool.dart

```diff
@@ -79,6 +79,10 @@ class _CanvasPool extends _SaveStackTracking {
     bool requiresClearRect = false;
     if (_reusablePool != null && _reusablePool.isNotEmpty) {
       _canvas = _reusablePool.removeAt(0);
+      // If a canvas is the first element we set z-index = -1 to workaround
+      // blink compositing bug. To make sure this does not leak when reused
+      // reset z-index.
+      _canvas.style.removeProperty('z-index');
```

Comment:
Wouldn't the effect be the same if you put this into an `else` branch of the `if (isFirstChildElement)` check below? Just so we have all the z-index related code together in one spot.

Reply:
No since it never reaches that code when you reuse from reuse_pool

Reply:
I'm referring to line 127. It does reach it. I don't see any early `return` statement. Am I missing something?

Reply:
Ah got it. So yes thats a good option as well. But this way we are executing the removeProperty call only if we reuse instead of doing it at startup as well.

Reply:
Ah, makes sense.